core/utils/samples.py

```diff
@@ -78,205 +78,117 @@ def add_molecule_protocol_parameters(form_data):
+            unique_sample_id__isnull=True
+        ).exists():
+            sample["unique_sample_id"] = "AAA-0001"
+        else:
+            last_unique_value = (
+                core.models.Samples.objects.exclude(unique_sample_id__isnull=True)
+                .last()
+                .unique_sample_id
+            )
+            sample["unique_sample_id"] = increase_unique_value(last_unique_value)
+
+        # Check if patient code  already exists on database,
+        # If not if will be created giving a sequencial dummy value
+        if sample["patient_core"] != "":
+            patient_obj = check_patient_code_exists(sample["patient_core"])
+            if patient_obj is False:
+                # Define the new patient only Patient code is defined
+                patient_obj = create_empty_patient(sample["patient_core"])
+        else:
+            patient_obj = None
 
-    Args:
-        request (_type_): _description_
-        app_name (string): application name
+        sample["patient_core"] = patient_obj
 
-    Returns:
-        dictionary: if there are incompleted or already defined samples, it return error key
-            with their sample names and the jexcel data to allow user correct the wrong data.
-            If not issues are found return the converted data is returned
-    """
-    reg_user = request.user.username
-    excel_json_data = json.loads(request.POST["table_data"])
-    c_data = core.utils.common.jspreadsheet_to_dict(
-        core.core_config.HEADING_FOR_RECORD_SAMPLES, excel_json_data
-    )
+        # Check if sample project exist and generate de appropiate objetct
+        if sample["sample_project"] == "None":
+            sample["sample_project"] = None
+        else:
+            sample["sample_project"] = core.models.SampleProjects.objects.get(
+                sample_project_name__exact=sample["sample_project"]
+            )
 
-    # create different list depending on the sample faulty case
-    defined_samples = []
-    incomplete_sample = []
+        # If only recorded set sample to completed state
+        if sample["only_recorded"]:
+            sample["sample_state"] = "Completed"
+            sample["completed_date"] = datetime.datetime.now()
+        else:
+            sample["sample_state"] = "Pre-Defined"
 
-    for sample in c_data:
-        if sample["Sample Name"] == "":
-            continue
-        # check if sample is defined for the same user
-        if check_if_sample_already_defined(sample["Sample Name"], reg_user):
-            defined_samples.append(sample["Sample Name"])
-            continue
-        # check mandatory fields
-        if sample["Type of Sample"] == "":
-            incomplete_sample.append(sample["Sample Name"])
-            continue
-        if not check_mandatory_fields_included(
-            sample, sample["Type of Sample"], app_name
-        ):
-            incomplete_sample.append(sample["Sample Name"])
+        try:
```

Comment:
I had in mind that samples were not recorded until all samples were validated.

Reply:
the validation is done prior to this function, it is called after all samples where validated. I have to code the chunk when there are some incomplete samples to allow the user to change the data and revalidate.